packages/ts-api-react-forms/src/ApiAction.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { useCallback } from "react";
+import { z, ZodObject, ZodRawShape } from "zod";
+import { ApiError } from "@thunderstore/thunderstore-api";
+import { ApiEndpoint } from "@thunderstore/ts-api-react";
+import { useApiAction } from "./useApiAction";
+
+export interface ApiActionProps<
+  Schema extends ZodObject<Z>,
+  Result extends object,
+  Z extends ZodRawShape
+> {
+  schema: Schema;
+  endpoint: ApiEndpoint<z.infer<Schema>, Result>;
+  metaData: any;
```

Comment:
What ever happens to this whole file, as per my earlier reviews I would prefer a) better typing b) silencing the warning and adding a comment describing why it was necessary. I like this approach since writing the comment usually (not necessarily in this case) works as rubber ducking, and makes it obvious to others that the choice to use `any` was deliberate and necessary.

Needless to say, this comment applies to all these warnings.